File: src/main/java/MachTeacher/MachTeacher/dto/RegisterRequest.java
package MachTeacher.MachTeacher.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class RegisterRequest {

    
    @NotBlank
    @Size(max = 80)
    private String fullName;

    @NotBlank
    @Email
    @Size(max = 120)
    private String email;

    @NotBlank
    @Size(min = 6, max = 120)
    private String password;

    
    @Size(max = 20)
    private String role;

    
    @Size(max = 30)
    private String phone;

    @Size(max = 80)
    private String city;

    @Size(max = 80)
    private String country;

    
    @Size(max = 120)
    private String university;

    @Size(max = 120)
    private String career;

    @Size(max = 20)
    private String semester;
    private String bio;
    private String subjects;
    private String hourlyRate;
    private String teachingExperience;
    private String aboutMe;
    private String references;
}

File: MachTeacher/app/src/main/java/com/example/machteacher/dto/RegisterRequest.kt
package com.example.machteacher.dto

data class RegisterRequest(

    val fullName: String,
    val email: String,
    val password: String,
    val role: String?,
    val phone: String? = null,
    val city: String? = null,
    val country: String? = null,
    // comunes
    val university: String? = null,
    val career: String? = null,
    val semester: String? = null,
    // mentor
    val bio: String? = null,
    val subjects: String? = null,
    val hourlyRate: String? = null,
    val teachingExperience: String? = null,
    val aboutMe: String? = null,
    val references: String? = null
)

